### ISSUE TITLE
demo.ckan.org fails when using datastore_search

### ISSUE THREAD

**Issue by frbattid** [ckan/ckan#4270]
### CKAN Version if known (or site URL)
I'm using CKAN running at `demo.ckan.org`.

### Please describe the expected behaviour
Once a resource has been uploaded to CKAN, I expect to search for data using CKAN REST API. Specifically, `datastore_search`.

### Please describe the actual behaviour
CKAN always return:

```
{
    "error": {
        "__type": "Not Found Error",
        "message": "Not found: Resource \"resource-id\" was not found."
    },
    "help": "https://demo.ckan.org/api/3/action/help_show?name=datastore_search",
    "success": false
}
```

### What steps can be taken to reproduce the issue? 
At `demo.ckan.org` create a dataset under an organization, then upload a CSV file as a resource, and query for it using `datastore_search` REST API operation.

**Comment by amercader** [comment]
`datastore_search` seems to be working fine: 

https://demo.ckan.org/api/3/action/datastore_search?resource_id=04127ad5-77e5-4a08-9f40-12d3c383e460&limit=5

@frbattid Are you replacing `resource-id` with the actual resource id (ie a UUID)? 
Is it happening with a particular file? Perhaps it contains an error and the file is not being uploaded to the DataStore.

**Comment by frbattid** [comment]
Thanks for replying.

For instance, it happens with this resource: https://demo.ckan.org/dataset/usal-test-dataset-error/resource/9f10a482-dc79-46fa-8a0c-494a592c442f

By the way, CKAN yesterday was able to visualize it, but not today.

The REST API tells me:

```
$ curl -X GET "https://demo.ckan.org/api/3/action/datastore_search?resource_id=9f10a482-dc79-46fa-8a0c-494a592c442f" -H "Authorization: xxxxxxxxxx"  -H "Content-Type: application/json" | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   226    0   226    0     0    217      0 --:--:--  0:00:01 --:--:--   217
{
    "error": {
        "__type": "Not Found Error",
        "message": "Not found: Resource \"9f10a482-dc79-46fa-8a0c-494a592c442f\" was not found."
    },
    "help": "https://demo.ckan.org/api/3/action/help_show?name=datastore_search",
    "success": false
}
```

Nevertheless, I agree it works for other resources I have.

**Comment by amercader** [comment]
There's nothing wrong with the file, I've downloaded the resource and uploaded to my local instance and it worked fine (the DataPusher uploaded it to the DataStore). Going to Resource page > Manage > DataStore will give you the error that the DataPusher is returning.
I can reproduce the error on demo, it's returning a 403 error, so I suspect it's something related to the particular setup of the file uploads on https://demo.ckan.org and the storage backend it is using (it also happens on https://beta.ckan.org, hosted on the same cluster).

So basically this should work fine on any other CKAN instance :)
